### PR TITLE
[WIP] JBIDE-26900 - Fix the failing maven.itests on Java 11 due to some timeouts

### PR DESCRIPTION
**Pull request by zcervink**
JBIDE-26900 - Fix the failing maven.itests on Java 11

**JIRA:** https://issues.redhat.com/browse/JBIDE-26900
**JENKINS:** https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/maven.itests/1066/

Please review @jkopriva @odockal

### CODE REVIEW

**Comment by zcervink** [comment]
@odockal @jkopriva  here is the line of code, that make issues on JDK 1.8 on mac:
`double version = Double.parseDouble(System.getProperty("java.specification.version"));`

Probably we should fix it, but I do not know how. Do you have any idea? (Of course, I am gonna fix the sonar issue too).

**Comment by zcervink** [comment]
Update - I have fixed the silly sonar issue. Now I am trying to resolve the mac Java 8 issue.

**Comment by zcervink** [comment]
testPR

File: tests/org.jboss.tools.maven.ui.bot.test/src/org/jboss/tools/maven/ui/bot/test/conversion/MavenConversionTest.java [hunk 1]
```diff
@@ -263,4 +267,16 @@ private void finishConversionDialog(){
 		}
 		return libs;
 	}
+	
+	public void replaceInPom(String target, String replacement, boolean save) {
```
Comment (jkopriva) [inline tests/org.jboss.tools.maven.ui.bot.test/src/org/jboss/tools/maven/ui/bot/test/conversion/MavenConversionTest.java]: Where is this used?

Comment (zcervink) [reply tests/org.jboss.tools.maven.ui.bot.test/src/org/jboss/tools/maven/ui/bot/test/conversion/MavenConversionTest.java]: Well, the PR has been edited several times due to some resolved issue and due to newly appeared issues. I must have created/copied this method and I must have forgotten to delete it when I come up to the conclusion that I do not need it. Deleted.

**Comment by zcervink** [comment]
@jkopriva @odockal I have spent some time on fixing the maven issues. Several timeout maven issues has been fixed, other issues are sporadically appearing, simple 1 or 2 reruns would cause the test to pass. Issues that blocks some tests and needs to be reported has been reported and fixed during working on this issue. I suggest to merge, other issue I am not able to fix right now.

**Jenkins:** https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/maven.itests/1136/

**Comment by zcervink** [comment]
@jkopriva I have resolved the conflict with the merged JBIDE-27146 branch and have also rebased. Now the PR is with the WIP label, I am going to work on it later.

**Comment by zcervink** [comment]
@jkopriva issue has almost disappeared over time, I recommend to close this PR as well as the JIRA (now the only issue in maven itests are the 4 failing tests we let to fail intentionally (we have discussed that a week ago during fixing the "checkbox" issue)). Sometimes there is still timeout issue, but it appears only sometime and with both J8 and J11, after rerun the issue is gone.

**Jenkins from today:**
- https://studio-jenkins-csb-codeready.cloud.paas.psi.redhat.com/job/Studio/job/Quality/job/Integration-tests/job/maven-itests/55/

**Pull request closed without merging**